### DISCUSSION TITLE
How to enable function hooks for existing projects?

### DISCUSSION THREAD

**Discussion by vimtor** [supabase/supabase#7901] [Questions] [answered]
I am implementing a way to minimize images after they are uploaded. I would love to setup a function hook that does a HTTP request to my Vercel serverless function which uses [`sharp`](https://www.npmjs.com/package/sharp) to resize the image and replace the original image with the minified one.

I've tried using SQL triggers and the [`http`](https://github.com/pramsey/pgsql-http) library, but they make the inserts too slow far since they happen synchronously. I've watched the [Supabase video about Function Hooks](https://www.youtube.com/watch?v=codAs9-NeHM) and it looks really promising to solve my situation. However I've read in the [release page](https://supabase.com/blog/2021/07/30/supabase-functions-updates) that it will be supported for _all **new** projects_.

I think I would have to [Zapier](https://zapier.com/) or something like that to be able to accomplish this kind of async trigger. I also thought about using a [Supabase edge functions](https://supabase.com/docs/guides/functions) but I did not find a way to subscribe to table inserts in the documentation.

**Comment by monicakh** [comment, marked as answer]
This feature is tied up with major version updates for Postgres and It seems you have an older version of the project PostgresSQL 13.3. To get access to Function Hooks you should migrate to new project.

Here is our guide for migrating from one Supabase project to another:
https://supabase.com/docs/guides/database#migrating-between-projects

​Feel free to let us know if you run into any trouble while performing the migration.
